src/Interpreters/ProcessList.h

```diff
@@ -301,11 +306,11 @@ class ProcessList
     size_t max_select_queries_amount = 0;
 
     /// amount of queries by query kind.
-    QueryKindToAmount query_kind_amounts;
+    QueryKindAmounts * query_kind_amounts = new QueryKindAmounts{0, 0};
```

Comment:
```suggestion
    std::unique_ptr<QueryKindAmounts> query_kind_amounts = std::make_unique<QueryKindAmounts>(0, 0);
```

Reply:
Thank you, I am going to return `unsorted_map<IAST::QueryKind, QueryAmount>` here